src/notation/view/widgets/editstaff.cpp

```diff
@@ -213,7 +213,12 @@ void EditStaff::updateInstrument()
 
     longName->setPlainText(m_instrument.nameAsPlainText());
     shortName->setPlainText(m_instrument.abbreviatureAsPlainText());
-    instrumentName->setText(m_instrument.nameAsPlainText());
+    const InstrumentTemplate* templ = mu::engraving::searchTemplate(m_instrument.id());
+    if (templ && !templ->musicXMLid.isEmpty()) {
+        instrumentName->setText(formatInstrumentTitle(templ->trackName, templ->trait));
+    } else {
+        instrumentName->setText(m_instrument.trackName());
```

Comment:
I think it would probably best to display "Unknown" (translated string) when there's no template. The goal of this field is to tell the user what instrument MuseScore *thinks* is there, which could affect certain behaviours like playback, engraving, and pitch range warnings. If MuseScore doesn't know what the instrument is then the user needs to be told this so that they can fix it by choosing one from the dialog.

Once implemented, we won't have a translation for "Unknown", but you'll be able to test you implemented it properly by downloading the CI build and changing the language to "Placeholder translations". If you see `tr<<Unknown>>` in the program then it means translations will work once we have translations for it.
![image](https://user-images.githubusercontent.com/11011881/229554574-3ae64699-4afe-4660-862c-845ce8cfc92a.png)